### ISSUE TITLE
maven plugin: ArrayStoreException: sun.reflect.annotation.TypeNotPresentExceptionProxy

### ISSUE THREAD

**Issue by bseiller** [EvoSuite/evosuite#5]
Trying to run 
`mvn compile evosuite:generate`
on a complex project results in this exception:
```
[INFO] * EvoSuite 1.0.1
[INFO] [MASTER] 10:42:31.430 [main] ERROR EvoSuite - Fatal crash on main EvoSuite process. Class  using seed 1448012551427. Configuration id : null
[INFO] java.lang.ArrayStoreException: sun.reflect.annotation.TypeNotPresentExceptionProxy
[INFO]  at sun.reflect.annotation.AnnotationParser.parseClassArray(AnnotationParser.java:724) ~[na:1.8.0_60]
[INFO]  at sun.reflect.annotation.AnnotationParser.parseArray(AnnotationParser.java:531) ~[na:1.8.0_60]
[INFO]  at sun.reflect.annotation.AnnotationParser.parseMemberValue(AnnotationParser.java:355) ~[na:1.8.0_60]
[INFO]  at sun.reflect.annotation.AnnotationParser.parseAnnotation2(AnnotationParser.java:286) ~[na:1.8.0_60]
[INFO]  at sun.reflect.annotation.AnnotationParser.parseAnnotations2(AnnotationParser.java:120) ~[na:1.8.0_60]
[INFO]  at sun.reflect.annotation.AnnotationParser.parseAnnotations(AnnotationParser.java:72) ~[na:1.8.0_60]
[INFO]  at java.lang.reflect.Executable.declaredAnnotations(Executable.java:602) ~[na:1.8.0_60]
[INFO]  at java.lang.reflect.Executable.declaredAnnotations(Executable.java:600) ~[na:1.8.0_60]
[INFO]  at java.lang.reflect.Executable.getDeclaredAnnotations(Executable.java:591) ~[na:1.8.0_60]
[INFO]  at java.lang.reflect.Method.getDeclaredAnnotations(Method.java:629) ~[na:1.8.0_60]
[INFO]  at java.lang.reflect.AccessibleObject.getAnnotations(AccessibleObject.java:207) ~[na:1.8.0_60]
[INFO]  at org.junit.runners.model.FrameworkMethod.getAnnotations(FrameworkMethod.java:187) ~[evosuite-master-1.0.1.jar:1.0.1]
[INFO]  at org.junit.runners.model.TestClass.addToAnnotationLists(TestClass.java:84) ~[evosuite-master-1.0.1.jar:1.0.1]
[INFO]  at org.junit.runners.model.TestClass.scanAnnotatedMembers(TestClass.java:66) ~[evosuite-master-1.0.1.jar:1.0.1]
[INFO]  at org.junit.runners.model.TestClass.<init>(TestClass.java:57) ~[evosuite-master-1.0.1.jar:1.0.1]
[INFO]  at org.evosuite.junit.CoverageAnalysis.isTest(CoverageAnalysis.java:632) ~[evosuite-master-1.0.1.jar:1.0.1]
[INFO]  at org.evosuite.continuous.project.ProjectAnalyzer.getCutsToAnalyze(ProjectAnalyzer.java:169) ~[evosuite-master-1.0.1.jar:1.0.1]
[INFO]  at org.evosuite.continuous.project.ProjectAnalyzer.analyze(ProjectAnalyzer.java:199) ~[evosuite-master-1.0.1.jar:1.0.1]
[INFO]  at org.evosuite.continuous.ContinuousTestGeneration.execute(ContinuousTestGeneration.java:155) ~[evosuite-master-1.0.1.jar:1.0.1]
[INFO]  at org.evosuite.executionmode.Continuous.execute(Continuous.java:110) ~[evosuite-master-1.0.1.jar:1.0.1]
[INFO]  at org.evosuite.EvoSuite.parseCommandLine(EvoSuite.java:259) ~[evosuite-master-1.0.1.jar:1.0.1]
[INFO]  at org.evosuite.EvoSuite.main(EvoSuite.java:295) ~[evosuite-master-1.0.1.jar:1.0.1]
[ERROR] Error in EvoSuite
...
[ERROR] Failed to execute goal org.evosuite.plugins:evosuite-maven-plugin:1.0.1:generate (default-cli) on project XYZ: Failed to correctly execute EvoSuite -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.evosuite.plugins:evosuite-maven-plugin:1.0.1:generate (default-cli) on project inverterComputationEJB: Failed to correctly execute EvoSuite
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:862)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:286)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:197)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoFailureException: Failed to correctly execute EvoSuite
        at org.evosuite.maven.GenerateMojo.runEvoSuiteOnSeparatedProcess(GenerateMojo.java:217)
        at org.evosuite.maven.GenerateMojo.execute(GenerateMojo.java:184)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
        ... 20 more
```
The issue seems to be that there are classes being scanned annotated with annotations whose definitions can't be found in the classpath.
For similar issues see here:
https://github.com/spring-projects/spring-boot/issues/1696
http://javahowto.blogspot.ca/2008/10/solve-javalangarraystoreexception.html

While the root cause of this might not be an error on itself (doing this via the command line works fine though) there is currently no way to tell which class contains which annotation that can't be found. I tried debugging the plugin but couldn't get any more information.
Catching this error and adding some details/context to the output would be very helpful :-)

**Comment by arcuri82** [comment]
I added a check on that exception, which now should contain the name of the class for which loading the annotations fails.

You ll need to pull latest EvoSuite from Git, and run

mvn install

then, you can modify your pom file to point to the current version, 1.0.2-SNAPSHOT

**Comment by bseiller** [comment]
Thanks - works like a charm!
Also there isn't this 
`ArrayStoreException: sun.reflect.annotation.TypeNotPresentExceptionProxy`
any more. AFIAK we did not change out project in any way - might it be one of the other changes in the 1.0.2-SNAPSHOT did that?

**Comment by arcuri82** [comment]
yes, few bugs have been fixed since we made release 1.0.1 some months ago. so likely that was one of them
